application/data/standardisers/ethnicity_classification_finder.py

```diff
@@ -0,0 +1,287 @@
+            "parent": self.parent,
+            "order": self.order,
+            "required": self.required,
+        }
+
+
+class EthnicityClassification:
+    def __init__(self, code, name):
+        self.code = code
+        self.name = name
+        self.standard_value_to_display_value_map = {}
+        self.classification_data_items = {}
+
+    def get_code(self):
+        return self.code
+
+    def get_name(self):
+        return self.name
+
+    def is_valid_for_raw_ethnicities(self, raw_ethnicities, ethnicity_standardiser):
+        standard_ethnicities_in_data = ethnicity_standardiser.standardise_all(raw_ethnicities)
+
+        return self.is_valid_for_standard_ethnicities(standard_ethnicities_in_data)
+
+    def is_valid_for_standard_ethnicities(self, standard_ethnicities):
+        unique_ethnicities = EthnicityClassification.__remove_duplicates(standard_ethnicities)
+
+        if self.__no_unknown_values(unique_ethnicities):
+            return self.__has_data_for_all_required_display_ethnicities(unique_ethnicities)
+        else:
+            return False
+
+    def add_data_item_to_classification(self, standard, classification_data_item):
+        self.standard_value_to_display_value_map[standard] = classification_data_item.display_ethnicity
+        self.classification_data_items[classification_data_item.display_ethnicity] = classification_data_item
+
+    def __has_data_for_all_required_display_ethnicities(self, standard_ethnicity_list):
+        required = self.__get_required_display_ethnicities()
+        display_ethnicity_list = [
+            self.standard_value_to_display_value_map[standard] for standard in standard_ethnicity_list
+        ]
+        for ethnicity in required:
+            if ethnicity not in display_ethnicity_list:
+                return False
+        return True
+
+    def __no_unknown_values(self, standard_ethnicity_list):
+        for ethnicity in standard_ethnicity_list:
+            if ethnicity not in self.standard_value_to_display_value_map:
+                return False
+        return True
+
+    def __get_required_display_ethnicities(self):
+        return {
+            classification_item.display_ethnicity
+            for classification_item in self.classification_data_items.values()
+            if classification_item.required is True
+        }
+
+    def __get_optional_display_ethnicities(self):
+        return {
+            classification_item.display_ethnicity
+            for classification_item in self.classification_data_items.values()
+            if classification_item.required is True
+        }
+
+    def get_data_fit_level(self, raw_ethnicities, standardiser):
+        true_values = 0
+        for raw_ethnicity in raw_ethnicities:
+            standard_ethnicity = standardiser.standardise(raw_ethnicity)
+            if standard_ethnicity in self.classification_data_items:
+                true_values += 1
+        return true_values
```

Comment:
FYI: This could potentially be written more concisely using set intersection:

```
standardised_ethnicities = {standardiser.standardise(raw_ethicity) for raw_ethnicity in raw_ethnicities}
return len(set(self.classification_data_items).intersection(standardised_ethnicities))
```